python/ray/tune/tests/test_function_api.py

```diff
@@ -96,163 +78,34 @@ def train(config, checkpoint_dir=None):
             new_trainable.stop()
         assert result[TRAINING_ITERATION] == 10
 
-    def testCheckpointReuseWithoutTraining(self):
-        """Test that repeated save/restore never reuses same checkpoint dir."""
-
-        def train(config, checkpoint_dir=None):
-            if checkpoint_dir:
-                count = sum(
-                    "checkpoint-" in path for path in os.listdir(checkpoint_dir)
-                )
-                assert count == 1, os.listdir(checkpoint_dir)
-
-            for step in range(20):
-                with tune.checkpoint_dir(step=step) as checkpoint_dir:
-                    path = os.path.join(checkpoint_dir, "checkpoint-{}".format(step))
-                    open(path, "a").close()
-                tune.report(test=step)
-
-        wrapped = wrap_function(train)
-        new_trainable = wrapped(logger_creator=self.logger_creator)
-        for i in range(2):
-            result = new_trainable.train()
-        checkpoint = new_trainable.save()
-        new_trainable.stop()
-
-        new_trainable2 = wrapped(logger_creator=self.logger_creator)
-        new_trainable2.restore(checkpoint)
-        new_trainable2.stop()
-
-        new_trainable2 = wrapped(logger_creator=self.logger_creator)
-        new_trainable2.restore(checkpoint)
-        result = new_trainable2.train()
-        new_trainable2.stop()
-        self.assertTrue(result[TRAINING_ITERATION] == 3)
-
-    def testReuseNullCheckpoint(self):
-        def train(config, checkpoint_dir=None):
-            assert not checkpoint_dir
-            for step in range(10):
-                tune.report(test=step)
-
-        # Create checkpoint
-        wrapped = wrap_function(train)
-        checkpoint = None
-        new_trainable = wrapped(logger_creator=self.logger_creator)
-        new_trainable.train()
-        checkpoint = new_trainable.save()
-        new_trainable.stop()
-
-        # Use the checkpoint a couple of times
-        for i in range(3):
-            new_trainable = wrapped(logger_creator=self.logger_creator)
-            new_trainable.restore(checkpoint)
-            new_trainable.stop()
-
-        # Make sure the result is still good
-        new_trainable = wrapped(logger_creator=self.logger_creator)
-        new_trainable.restore(checkpoint)
-        result = new_trainable.train()
-        checkpoint = new_trainable.save()
-        new_trainable.stop()
-        self.assertTrue(result[TRAINING_ITERATION] == 1)
-
-    def testMultipleNullCheckpoints(self):
-        def train(config, checkpoint_dir=None):
-            assert not checkpoint_dir
-            for step in range(10):
-                tune.report(test=step)
```

Comment:
Null checkpoints are created if you call `save` on a function trainable without ever reporting a checkpoint. This is not a real usage of the API, and "null checkpoints" are no longer created.